### DISCUSSION TITLE
Python request - I can't believe it.

### DISCUSSION THREAD

**Discussion by Makebuildmodify** [etsy/open-api#812] [Q&A] [answered]
Hi, this is what I have.

```
	etsy_sku_response = requests.put(
		url = update_sku_url,
		headers = {"x-api-key": "xxxxxxxxxxxxxxxxx","authorization": "Bearer xxxxxxxxxxxxxxxx"},
		json = products_json,
	)
```

And I'm getting 

STATUS CODE  AND STATUS TEXT--------------->  400, API request missing api_key or valid OAuth parameters

What am I missing? 

P.S. All of my other request.put()'s of the same form are working just fine. It's maddening.

Also, my cURL request works just fine with the same URL, headers, and JSON

`curl -X PUT https://openapi.etsy.com/v3/application/listings/1390807615/inventory -H "Content-Type: application/json" --data '{"products": [{"sku": "999999", "offerings": [{"quantity": 1, "is_enabled": true, "price": "9.00"}]}]}' -H 'x-api-key: xxxxxxxxxxxxxx' -H 'authorization: Bearer xxxxxxxxxxxxx' -w "Type: %{response_code}\n"`

**Comment by Makebuildmodify** [comment, marked as answer]
Nevermind, I solved it.

**Reply by baahar** [reply]
I'm not using Python and so couldn't help, but am curious what the issue was :)

**Reply by Makebuildmodify** [reply]
@baahar 
It's was a simple mistake in my code. I spent 16+ hours of reading API and HTTP request documentation to no avail before I found my mistake. 

I was changing the old API v2 functions into API v3 functions one at a time. So, I had both endpoints available in my script. Up to that point all of the new function  were working fine except for this one. It turns out that I my URL variable was pointing to an old API v2 endpoint. This is why the server response error wasn't making any sense to me. I was reading the error in the context of the API v3 error messages.  It's embarrassingly obvious now.

**Reply by baahar** [reply]
Ah, I see. Who hasn't spent hours and hours on problems like that :) Glad it is resolved now.

**Reply by Makebuildmodify** [reply]
I find that these kinds of problems are the best teachers. I never forget the lesson.
